### DISCUSSION TITLE
transformer for live image filtering.

### DISCUSSION THREAD

**Discussion by farfromrefug** [SDWebImage/SDWebImage#3614] [Q&A] [answered]
I am looking at how to implement image filtering (b&w, enhance...) with sdwebimage. I am thinking transformers are the way to go.
now I see how to use a transformer while loading an image, but is there a way to "change the transformers" without going through the process of making another request? like I am thinking about something fast where the original image would remain in memory and then changing transformer would be pretty fast too. I am Aldo hoping cachig would work too
is there a way (maybe another one) to do that ?

**Comment by dreampiggy** [comment, marked as answer]
> but is there a way to "change the transformers" without going through the process of making another request

Can you explain in detail about your feature request ?

Actually, SDWebImage store the original image data into disk (not memory), see the default value of `SDWebImageContextOriginalStoreCacheType`

So, if you change transformer with a new request, it will just hit disk cache and do decoding and transforming on CPU (no network request), and give you the transformed image. Different transformed image store into different memory cache and disk cache with the proper key (see the default value of `SDWebImageContextStoreCacheType`)

### The transformer and cache system design is more complicated than you think. Check the source code as well.

**Reply by farfromrefug** [reply]
@dreampiggy thanks i understand what you mean, and it is exactly what i want.
